test/functional/mempool_fee_histogram.py

```diff
@@ -0,0 +1,117 @@
+        utxos = node.listunspent()
+        assert_equal(2, len(utxos))
+        node.lockunspent(False, [{"txid": utxos[1]["txid"], "vout": utxos[1]["vout"]}])
+
+        self.log.info("Send tx1 transaction with 5 sat/vB fee rate")
+        node.sendtoaddress(address=node.getnewaddress(), amount=Decimal("50.0"), fee_rate=5, subtractfeefromamount=True)
+
+        self.log.info("Test fee rate histogram when mempool contains 1 transaction (tx1: 5 sat/vB)")
+        info = node.getmempoolinfo([1, 3, 5, 10])
+        (non_empty_groups, empty_groups, total_fees) = self.histogram_stats(info['fee_histogram'])
+        assert_equal(1, non_empty_groups)
+        assert_equal(3, empty_groups)
+        assert_equal(1, info['fee_histogram']['fee_rate_groups']['5']['count'])
+        assert_equal(total_fees, info['fee_histogram']['total_fees'])
+
+        self.log.info("Send tx2 transaction with 14 sat/vB fee rate (spends tx1 utxo)")
+        node.sendtoaddress(address=node.getnewaddress(), amount=Decimal("25.0"), fee_rate=14)
+
+        self.log.info("Test fee rate histogram when mempool contains 2 transactions (tx1: 5 sat/vB, tx2: 14 sat/vB)")
+        info = node.getmempoolinfo([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15])
+
+        # Both tx1 and tx2 are supposed to be reported in 8 sat/vB fee rate group
+        (non_empty_groups, empty_groups, total_fees) = self.histogram_stats(info['fee_histogram'])
+        assert_equal(1, non_empty_groups)
+        assert_equal(14, empty_groups)
+        assert_equal(2, info['fee_histogram']['fee_rate_groups']['8']['count'])
+        assert_equal(total_fees, info['fee_histogram']['total_fees'])
+
+        # Unlock the second UTXO which we locked
+        node.lockunspent(True, [{"txid": utxos[1]["txid"], "vout": utxos[1]["vout"]}])
+
+        self.log.info("Send tx3 transaction with 6 sat/vB fee rate (spends all available utxos)")
+        node.sendtoaddress(address=node.getnewaddress(), amount=Decimal("99.9"), fee_rate=6)
+
+        self.log.info("Test fee rate histogram when mempool contains 3 transactions (tx1: 5 sat/vB, tx2: 14 sat/vB, tx3: 6 sat/vB)")
+        info = node.getmempoolinfo([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15])
+
+        # Verify that each of 6, 7 and 8 sat/vB fee rate groups contain one transaction
+        (non_empty_groups, empty_groups, total_fees) = self.histogram_stats(info['fee_histogram'])
+        assert_equal(3, non_empty_groups)
+        assert_equal(12, empty_groups)
+        assert_equal(1, info['fee_histogram']['fee_rate_groups']['6']['count'])
+        assert_equal(1, info['fee_histogram']['fee_rate_groups']['7']['count'])
+        assert_equal(1, info['fee_histogram']['fee_rate_groups']['8']['count'])
+        assert_equal(total_fees, info['fee_histogram']['total_fees'])
+
+
+    def histogram_stats(self, histogram):
+        total_fees = 0
+        empty_count = 0
+        non_empty_count = 0
+
+        for key, bin in histogram['fee_rate_groups'].items():
```

Comment:
Is `bin` reserved in Python?

Reply:
It is not according to https://docs.python.org/3/reference/lexical_analysis.html#keywords